worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -717,7 +717,7 @@ private[celeborn] class Worker(
       applicationId: String = null,
       resourceConsumption: ResourceConsumption = null): Unit = {
     var resourceConsumptionLabel = userIdentifier.toMap
-    if (applicationId != null)
+    if (applicationId != null && conf.metricsPushApplicationId)
       resourceConsumptionLabel += (resourceConsumptionSource.applicationLabel -> applicationId)
```

Comment:
It should be fine to gauge the top application consumption?

1. `celeborn.metrics.worker.app.topResourceConsumption.count` is 50 by defaults
2. with https://github.com/apache/celeborn/pull/3232, I can set `celeborn.metrics.worker.app.topResourceConsumption.bytesWrittenThreshold` to 1G to limit the top app metrics

Reply:
You can also set `celeborn.metrics.worker.app.topResourceConsumption.count` to 0 disable the top app metrics.

Reply:
Replied here – https://github.com/apache/celeborn/pull/3221#discussion_r2063353314
> if we remove this default behaviour will be send top app consumption with application label, which will break the resource consumption with userIdentifier as well.
> IMO we should keep it or set the top resource consumption count default value to 0 + add a note about cardinality in the config documentation. WDYT?